dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoaderRef.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.classloading.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.security.ProtectionDomain;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Enumeration;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.framework.Bundle;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.classloading.LibertyClassLoader;
+import com.ibm.wsspi.classloading.ApiType;
+import com.ibm.wsspi.classloading.ClassLoaderIdentity;
+
+/**
+ * This class is used to create weak references to AppClassLoaders
+ */
+@Trivial
+public class ClassLoaderRef extends LibertyLoader implements SpringLoader {
+    static {
+        ClassLoader.registerAsParallelCapable();
+    }
+
+    private static final TraceComponent tc = Tr.register(ClassLoaderRef.class);
+
+    private static final Set<String> warnedAppNames = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    private final String key;
+
+    private final WeakReference<AppClassLoader> classLoaderWeakRef;
+
+    ClassLoaderRef(AppClassLoader appClassLoader) {
+        super(appClassLoader.getParent());
+        ClassLoaderIdentity identity = appClassLoader.getKey();
+        key = identity == null ? null : identity.toString();
+        classLoaderWeakRef = new WeakReference<>(appClassLoader);
+    }
+
+    @Override
+    protected final Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        return loadClass(name, resolve, false, false);
+    }
+
+    /**
+     * Returns the application class loader if it is still referenced
+     * and puts out a warning one time if it is not referenced any longer 
+     *
+     * @return the AppClassLoader or null
+     */
+    private AppClassLoader getAppLoader() {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        if (appCL == null) {
+            if (key != null && warnedAppNames.add(key)) {
+                Tr.warning(tc, "app.classloader.removed", Thread.currentThread().getName(), key);
```

Comment:
Did you intend to include the message strings for the `app.classloader.removed` key in this PR yet?